src/Core/Checkout/Customer/Validation/Constraint/CustomerZipCode.php

```diff
@@ -18,44 +19,44 @@ class CustomerZipCode extends Constraint
         self::ZIP_CODE_INVALID => 'ZIP_CODE_INVALID',
     ];
 
-    public ?string $countryId = null;
-
-    public bool $caseSensitiveCheck = true;
-
     private string $message = 'This value is not a valid ZIP code for country {{ iso }}';
 
     private string $messageRequired = 'Postal code is required for that country';
 
     /**
      * @param ?array{countryId?: ?string, caseSensitiveCheck?: bool} $options
      *
-     * @deprecated tag:v6.8.0 - Parameter $options will be required and natively typed as array
+     * @deprecated tag:v6.8.0 - Parameter $options will be removed
      */
-    public function __construct($options = null)
+    #[HasNamedArguments]
+    public function __construct($options = null, public bool $caseSensitiveCheck = true, public ?string $countryId = null)
```

Comment:
just out of curiosity: why are the properties here public and not protected like in all the other constraints?

Reply:
deprecate them and add changelog (tbh change the access modifier is not possible programmatically (unless you need to define a magic getter method and throw exception if they use the properties publicly)